app/controllers/feedbacks_controller.rb

```diff
@@ -9,9 +9,8 @@ def new
 
   def create
     @feedback = Feedback.new(feedback_params)
-    if visit = session[:visit]
-      @feedback.prison = visit.prisoner.prison_name
-    end
+    @feedback.prison = prison_name if visit
```

Comment:
I think you've accidentally dropped the assignment here, so `visit` will always be `nil`.